frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,256 @@
+import { Link } from 'react-router-dom';
+import IconWithTooltip from 'src/atoms/IconWithTooltip';
+import { color, commonCss, padding } from 'src/Css';
+import { ErrorBoundary } from 'src/lib/ErrorBoundary';
+import {
+  ExecutionHelpers,
+  filterArtifactsByType,
+  getArtifactTypes,
+  getOutputArtifactsInExecution,
+} from 'src/lib/MlmdUtils';
+import Banner from '../Banner';
+import { RoutePageFactory } from '../Router';
+import ROCCurve, { ROCCurveConfig } from '../viewers/ROCCurve';
+import { PlotType } from '../viewers/Viewer';
+
+const ROC_CURVE_DEFINITION =
+  'The receiver operating characteristic (ROC) curve shows the trade-off between true positive rate and false positive rate. ' +
+  'A lower threshold results in a higher true positive rate (and a higher false positive rate), ' +
+  'while a higher threshold results in a lower true positive rate (and a lower false positive rate)';
+
+type ConfidenceMetric = {
+  confidenceThreshold: string;
+  falsePositiveRate: number;
+  recall: number;
+};
+
+type MetricsTabProps = {
+  execution: Execution;
+};
+
+interface MetricsSwitcherProps {
+  artifacts: Artifact[];
+  artifactTypes: ArtifactType[];
+  refresh?: () => void;
+}
+
+/**
+ * Metrics tab renders metrics for the artifact of given execution.
+ * Some system metrics are: Confusion Matrix, ROC Curve, Scalar, etc.
+ * Detail can be found in https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/io_types.py
+ * Note that these metrics are only available on KFP v2 mode.
+ */
+export function MetricsTab({ execution }: MetricsTabProps) {
+  const [refreshCount, setRefreshCount] = useState(0);
+  function refresh() {
+    setRefreshCount(refreshCount + 1);
+  }
+  const [enableFetch, setEnableFetch] = useState(false);
+  const executionState = execution.getLastKnownState();
+  if (
+    !enableFetch &&
+    !(
+      executionState === Execution.State.NEW ||
+      executionState === Execution.State.UNKNOWN ||
+      executionState === Execution.State.RUNNING
+    )
+  ) {
+    setEnableFetch(true);
+  }
+
+  // Retrieving a list of artifacts associated with this execution,
+  // so we can find the artifact for system metrics from there.
+  const {
+    isLoading: isLoadingArtifacts,
+    isSuccess: isSuccessArtifacts,
+    error: errorArtifacts,
+    data: artifacts,
+  } = useQuery<Artifact[], Error>(
+    ['execution_output_artifact', { id: execution.getId() }],
+    () => getOutputArtifactsInExecution(execution),
+    { enabled: enableFetch, staleTime: Infinity },
+  );
+
+  // artifactTypes allows us to map from artifactIds to artifactTypeNames,
+  // so we can identify metrics artifact provided by system.
+  const {
+    isLoading: isLoadingArtifactTypes,
+    isSuccess: isSuccessArtifactTypes,
+    error: errorArtifactTypes,
+    data: artifactTypes,
+  } = useQuery<ArtifactType[], Error>(['artifact_types'], () => getArtifactTypes(), {
+    enabled: enableFetch,
+    staleTime: Infinity,
+  });
```

Comment:
nit: this query might still experience a stale cache, because query args has only 'artifact_types' string.

If we visit a different page that queries 'artifact_types' and we come to this page after that, a stale cache might not include the latest artifact types we care about.

I'd suggest explicitly getting artifact types for those showed up in the pipeline, because the cluster might have many many artifact types than this pipeline actually used. Including list of artifact type IDs in query args also makes sure the stale cache problem is avoided, because the query will refetch any time we see a new type.

Reply:
I agree that the cache might be stale if we switch to different page. In this case, I think including the artifact type IDs in query args might not solve the problem: what if two pages have the same type IDs? And we need to wait for the `getArtifacts` finish first to get artifact ID list, then call `getArtifactTypes` using ID list. It might have some performance shortcoming. 

I suggest to use execution ID as part of the `queryKey`. So we will refetch when switching page, and enable parallel query between `getArtifacts` and `getArtifactTypes`. How do you think?

Reply:
This is a brilliant idea! It solves all the problems except my concern for scalability. I think we can start with this solution before trying something too complex. It's never too late to re-evaluate when we find real perf problems.

> I'd suggest explicitly getting artifact types for those showed up in the pipeline, because the cluster might have many many artifact types than this pipeline actually used.